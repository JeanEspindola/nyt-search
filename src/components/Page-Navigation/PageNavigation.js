import React, { Component } from 'react';
import { Pager, Glyphicon } from 'react-bootstrap';
import PropTypes from 'prop-types';
import WithLocale from '../HOC/WithLocale';

class PageNavigation extends Component {
  constructor(props) {
    super(props);
    this.getNewValues = this.getNewValues.bind(this);
  }

  getNewValues(e) {
    const index = e.target.name;
    e.preventDefault();
    const page = index === 'next' ? this.props.page + 1 : this.props.page - 1;
    this.props.onQuerySearch(this.props.query, page);
  }

  render() {
    return (
      <Pager>
        <Pager.Item
          previous
          disabled={this.props.page === 0}
          onClick={this.getNewValues}
          name="previous"
        >
          <Glyphicon glyph="chevron-left" />
          {this.props.locale.previous_page}
        </Pager.Item>
        <Pager.Item
          next
          onClick={this.getNewValues}
          name="next"
        >
          {this.props.locale.next_page}
          <Glyphicon glyph="chevron-right" />
        </Pager.Item>
      </Pager>
    );
  }
}

PageNavigation.propTypes = {
  page: PropTypes.number.isRequired,
  query: PropTypes.string.isRequired,
  onQuerySearch: PropTypes.func.isRequired,
  locale: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default WithLocale(PageNavigation);
