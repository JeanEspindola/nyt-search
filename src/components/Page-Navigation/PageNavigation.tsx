import * as Icon from 'react-bootstrap-icons'
import { useDispatch, useSelector } from 'react-redux'
import { FormattedMessage } from 'react-intl'
import { RootStateType } from '../../redux/rootTypes'
import { articleSearchSubmit } from '../../redux/articleSearch/articleSearchAction'
import styles from './PageNavigation.module.scss'
import cx from 'classnames'

const PageNavigation = () => {
  const { page, query } = useSelector((state: RootStateType) => state.articleList)
  const dispatch = useDispatch()

  const getNewValues = (action: string) => {
    const newPage = action === 'next' ? page + 1 : page - 1
    dispatch(articleSearchSubmit({ query, page: newPage }))
  }

  const isFirstPage = page === 0

  return (
    <div className={styles.container}>
      <button
        className={cx(styles.baseButton, {
          [styles.disabledButton]: isFirstPage,
          [styles.activeButton]: !isFirstPage,
        })}
        onClick={() => getNewValues('previous')}
        disabled={page === 0}>
        <Icon.ChevronLeft />
        <FormattedMessage id={'previousPage'} />
      </button>
      <button
        className={cx(styles.baseButton, styles.activeButton, styles.nextButton)}
        onClick={() => getNewValues('next')}>
        <FormattedMessage id={'nextPage'} />
        <Icon.ChevronRight />
      </button>
    </div>
  )
}

export default PageNavigation
