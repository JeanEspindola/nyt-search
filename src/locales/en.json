{
  "go_to_results": "Go to Results Page",
  "header_title": "\"The New York Times\" article search application",
  "next_page": "Next Page",
  "no_details": "No details to show.",
  "no_results": "There are no results to display.",
  "previous_page": "Previous Page",
  "read_article": "Read the Full Article",
  "results": "Results:",
  "search_terms": "Search your terms here.",
  "type_query": "Type search query term in here:"
}