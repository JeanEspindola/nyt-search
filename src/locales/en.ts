// eslint-disable-next-line import/no-anonymous-default-export
export default {
  goToResults: "Go to Results Page",
  headerTitle: "\"The New York Times\" article search application",
  nextPage: "Next Page",
  noDetails: "No details to show.",
  noResults: "There are no results to display.",
  previousPage: "Previous Page",
  readArticle: "Read the Full Article",
  results: "Results:",
  searchTerms: "Search your terms here.",
  typeQuery: "Type search query term in here:"
}
