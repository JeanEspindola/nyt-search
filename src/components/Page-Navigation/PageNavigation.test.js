import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import PageNavigation from './PageNavigation';
import en from '../../locales/en.json';

describe('PageNavigation', () => {
  const mockOnQuerySearch = jest.fn();
  const func = () => {};
  const page = 2;
  const query = 'search';
  let list;
  let Component;

  beforeEach(() => {
    list = [{ id: 1, headline: { main: 'a' } }, { id: 2, headline: { main: 'a' } }];

    Component = shallow(
      <PageNavigation
        list={list}
        page={page}
        query={query}
        onQuerySearch={mockOnQuerySearch}
      />,
      { context: { locale: en } },
    ).find('PageNavigation')
      .shallow();
  });

  it('renders PageNavigation without crashing', () => {
    expect(Component.length).toBeTruthy();
    expect(shallowToJson(Component)).toMatchSnapshot();
  });

  it('should call getNewValues and call onQuerySearch -- next', () => {
    const event = {
      preventDefault: func,
      target: {
        name: 'next',
      },
    };
    Component.instance().getNewValues(event);
    expect(mockOnQuerySearch).toHaveBeenCalledWith('search', 3);
  });

  it('should call getNewValues and call onQuerySearch -- next', () => {
    const event = {
      preventDefault: func,
      target: {
        name: 'previous',
      },
    };
    Component.instance().getNewValues(event);
    expect(mockOnQuerySearch).toHaveBeenCalledWith('search', 1);
  });
});
