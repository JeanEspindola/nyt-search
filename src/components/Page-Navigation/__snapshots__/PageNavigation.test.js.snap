// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNavigation renders PageNavigation without crashing 1`] = `
<Pager
  bsClass="pager"
>
  <PagerItem
    disabled={false}
    name="previous"
    next={false}
    onClick={[Function]}
    previous={true}
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="chevron-left"
    />
    Previous Page
  </PagerItem>
  <PagerItem
    disabled={false}
    name="next"
    next={true}
    onClick={[Function]}
    previous={false}
  >
    Next Page
    <Glyphicon
      bsClass="glyphicon"
      glyph="chevron-right"
    />
  </PagerItem>
</Pager>
`;
